### PR TITLE
260/user export load

### PR DESCRIPTION
**Pull request by pyphilia**

### CODE REVIEW

**Comment by pyphilia** [comment]
The screenshots for the export:
image
image
The resources and actions are checked if there exists in the database.

And for load:
image
image
The checkboxes are checked depending on the zip file, if it contains resources/actions. 
Also we uncheck space if it already exists (same space), and check it if it is different/doesn't exist. It is always disabled in order to keep data in sync.
For student, if the spaces are different, they cannot load anything (an error message is sent)

File: src/components/space/load/LoadSelectionScreen.js [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+    const { classes, t, activity, elements, extractPath } = this.props;
+    const {
+      space: isSpaceChecked,
+      appInstanceResources: isResourcesChecked,
+      actions: isActionsChecked,
+    } = this.state;
+
+    const selection = [isSpaceChecked, isResourcesChecked, isActionsChecked];
+
+    // if extractPath is undefined, it returns in componentDidUpdate
```
Comment (juancarlosfarah) [inline src/components/space/load/LoadSelectionScreen.js:233]: Not sure I understand the comment.

File: src/components/space/load/LoadSelectionScreen.js [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+                t(`This Space's Resources`),
+                isResourcesChecked,
+                elements.get('appInstanceResources').isEmpty(),
+                t(`This file does not contain any user input`)
+              )}
+            </Grid>
+            <Grid item xs={1}>
+              <Tooltip
+                title={t(
+                  'Resources are inputs a user save when using applications (ie. answer in Input Text App).'
```
Comment (juancarlosfarah) [inline src/components/space/load/LoadSelectionScreen.js:289]: Same change as in the other screen.

File: src/components/space/load/LoadSelectionScreen.js [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+                t(`This Space's Actions`),
+                isActionsChecked,
+                elements.get('actions').isEmpty(),
+                t(`This file does not contain any analytics`)
+              )}
+            </Grid>
+            <Grid item xs={1}>
+              <Tooltip
+                title={t(
+                  'Actions are various analytics and user data a user left while using Graasp Desktop.'
```
Comment (juancarlosfarah) [inline src/components/space/load/LoadSelectionScreen.js:308]: Same change as in the other screen.

File: src/reducers/exportSpaceReducer.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  CLEAR_EXPORT_SPACE,
+  SET_EXPORT_SPACE,
+  FLAG_SETTING_EXPORT_SPACE,
+  EXPORT_SPACE_SUCCESS,
+  FLAG_CLEARING_EXPORT_SPACE,
+} from '../types';
+import { updateActivityList } from './common';
+
+export const EXPORT_SPACE_STATUS = {
+  IDLE: 'idle',
```
Comment (juancarlosfarah) [inline src/reducers/exportSpaceReducer.js:13]: What does `idle` mean?

File: src/config/messages.js [hunk 5]
```diff
@@ -63,6 +63,9 @@ const ERROR_SETTING_SPACE_AS_FAVORITE =
   'There was an error setting space as favorite';
 const ERROR_SETTING_SPACE_AS_RECENT =
   'There was an error setting space as recent space';
+const ERROR_STUDENT_LOAD_OUT_OF_DATA_SPACE_MESSAGE =
```
Comment (juancarlosfarah) [inline src/config/messages.js:66]: Update message as above.

File: public/app/listeners/loadSpace.js [hunk 6]
```diff
@@ -47,62 +83,139 @@ const loadSpace = (mainWindow, db) => async (event, { fileLocation }) => {
...
+      const savedResources = db.get(APP_INSTANCE_RESOURCES_COLLECTION);
 
-    if (!wasRenamed) {
-      logger.error('unable to rename extract path');
-      mainWindow.webContents.send(LOADED_SPACE_CHANNEL, ERROR_GENERAL);
-      return clean(extractPath);
+      const newResources = appInstanceResources
+        // keep only non-duplicate resources
+        .filter(({ id }) => !savedResources.find({ id }).value())
+        // change user id by current user id
```
[the remark this answers is not in the document]
Comment (pyphilia) [reply public/app/listeners/loadSpace.js:184]: I added a warning banner 

image

Is the message clear enough?

File: src/components/space/export/ExportSelectionScreen.js [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+    component: 'span',
+    fontStyle: 'italic',
+
+    // following lines ensure long titles are shorten
+    display: 'inline-block',
+    maxWidth: 200,
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    whiteSpace: 'nowrap',
+
```
[the remark this answers is not in the document]
Comment (pyphilia) [reply src/components/space/export/ExportSelectionScreen.js:52]: It is really 1px which does not exist in `theme.spacing`, it is to create a blank space just before the space name.

File: src/components/space/export/ExportSelectionScreen.js [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+    marginLeft: 1, // simulate left space
```
[the remark this answers is not in the document]
Comment (pyphilia) [reply src/components/space/export/ExportSelectionScreen.js:53]: It is really 1px which does not exist in theme.spacing, it is to create a blank space just before the space name.

File: src/reducers/exportSpaceReducer.js (hunk 4 above)
Comment (pyphilia) [reply src/reducers/exportSpaceReducer.js:13]: It means it does nothing. It opposes to `pending` where the action is processing. Should I change it to something else?

**Pull request merged**